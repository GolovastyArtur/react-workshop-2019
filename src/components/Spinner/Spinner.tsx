import * as React from 'react';

import classnames from 'classnames';

import './Spinner.css';

export interface SpinnerProps {
    screen?: boolean;
    className?: string;
}

export const Spinner: React.FC<SpinnerProps & React.RefAttributes<HTMLDivElement>> = React.forwardRef(
    function Spinner({screen, className}: SpinnerProps, ref: React.Ref<HTMLDivElement>) {
        return (
            <div
                ref={ref}
                className={classnames('spinner', className, {
                    'spinner_screen': screen,
                })}
            >
                <div className="spinner__inner" />
            </div>
        );
    }
);

