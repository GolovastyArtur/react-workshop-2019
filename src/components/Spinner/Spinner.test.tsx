import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {Spinner} from './Spinner';

describe('Spinner', function () {
    it('render default', function() {
        let component = renderer.create(<Spinner />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('render screen mod', function() {
        let component = renderer.create(<Spinner screen />);

        expect(component.toJSON()).toMatchSnapshot();
    });
});
