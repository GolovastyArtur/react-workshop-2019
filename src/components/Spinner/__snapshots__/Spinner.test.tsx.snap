// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner render default 1`] = `
<div
  className="spinner"
>
  <div
    className="spinner__inner"
  />
</div>
`;

exports[`Spinner render screen mod 1`] = `
<div
  className="spinner spinner_screen"
>
  <div
    className="spinner__inner"
  />
</div>
`;
